### PR TITLE
As mcalling java

### PR DESCRIPTION
**Pull request by neilbeesley**
This is a performance improvement, by removing a lot of getmains/freemains and also using stack (CEEDSA) storage and staying in 64 bit mode whenever possible, which is the AMODE of Java.
Spellings also corrected and dead code removed.

### CODE REVIEW

File: POC-iterations/ASMcallingJAVA/ASM/asminf264.s [hunk 1]
```diff
@@ -379,9 +289,11 @@ PAADDR1  DS    D
 PAADDR2  DS    D 
 PARETC   DS    D
 PAANCHR  DS    D
+PAATMEM  DS    D
 PARMLEN  EQU   *-PARMSTR
 *
-         CEEDSA SECTYPE=XPLINK         Dynamic Save Area
+MYDSA    CEEDSA SECTYPE=XPLINK         Dynamic Save Area
```
Comment (KipTwitchell) [inline POC-iterations/ASMcallingJAVA/ASM/asminf264.s:295]: THEDSA    CEEDSA SECTYPE=XPLINK         Dynamic Save Area

File: POC-iterations/ASMcallingJAVA/ASM/asminf264.s [hunk 2]
```diff
@@ -2,33 +2,35 @@
...
 * ASMINF64 - JNI Interface to ASM service routines (64 bit)
 *
 ***********************************************************************
          YREGS                                                                  
 *                                                                               
-ASMINF64 CELQPRLG PARMWRDS=1,BASEREG=R10,EXPORT=YES                             
+ASMINF64 CELQPRLG PARMWRDS=1,BASEREG=R10,EXPORT=YES,DSASIZE=160+WORKLEN
          LG    R9,0(R1)                R9 -> Input area                         
          USING PARMSTR,R9
+         LAY   R13,MYWORK-MYDSA(,R4)
```
Comment (KipTwitchell) [inline POC-iterations/ASMcallingJAVA/ASM/asminf264.s:32]: LAY   R13,MYWORK-THEDSA(,R4)

File: POC-iterations/ASMcallingJAVA/ASM/gvbjmain.s [hunk 3]
```diff
@@ -41,7 +41,7 @@ SAVEAREA DC    18F'0'
 SAVER13  DS    D
 *
          DS    0F
-OUTDCB   DS    XL(OUTFILEL)    REENTRANT DCB AND DCBE AREAS
+OUTDCB   DS    A
```
Comment (KipTwitchell) [inline POC-iterations/ASMcallingJAVA/ASM/gvbjmain.s:44]: add comment that it is the address of the DCB

File: POC-iterations/ASMcallingJAVA/ASM/gvbjmain.s [hunk 4]
```diff
@@ -114,6 +114,7 @@ PAADDR1  DS    D
 PAADDR2  DS    D
```
Comment (KipTwitchell) [inline POC-iterations/ASMcallingJAVA/ASM/gvbjmain.s:114]: Add comments about each parameter

File: POC-iterations/ASMcallingJAVA/ASM/gvbjmr95.s [hunk 5]
```diff
@@ -294,14 +309,7 @@ MAIN_140 EQU   *
...
 *
-*        ALLOCATE TABLE for REQUEST communication: CTRAREA
-*
-MAIN_114 EQU   *
-** *     WTO 'GVBJMR95: COMMUNICATIONS TENSOR TABLE LOCATED'
-*
-*        LOAD AND EXECUTE GVBMR95
-*
-*         LOAD  EP=GVBMR95E
+MAIN_114 EQU   *                  LOAD AND EXECUTE GVBMR95 OR SOMETHING
```
Comment (KipTwitchell) [inline POC-iterations/ASMcallingJAVA/ASM/gvbjmr95.s:312]: add more prominent comment on this.

File: POC-iterations/ASMcallingJAVA/ASM/gvbjmr95.s [hunk 6]
```diff
@@ -310,29 +318,27 @@ MAIN_114 EQU   *
          LAY   R1,WKDDPRML
          ST    R1,WKEPARMA
          LAY   R1,WKEPARMA
-         L     R15,GVBMR95E
-         BASR  R14,R15
+         L     R15,GVBMR95E         This may not store 64 bit registers
+         BASR  R14,R15                                 only 31 bit !!!!
 *
          STIMER WAIT,BINTVL=FIVESEC
```
Comment (KipTwitchell) [inline POC-iterations/ASMcallingJAVA/ASM/gvbjmr95.s:324]: add comment about subprograms cleaning up.

File: POC-iterations/ASMcallingJAVA/ASM/gvbjpost.s [hunk 7]
```diff
@@ -114,6 +114,7 @@ PAADDR1  DS    D
 PAADDR2  DS    D
```
Comment (KipTwitchell) [inline POC-iterations/ASMcallingJAVA/ASM/gvbjpost.s:114]: Comments

File: POC-iterations/ASMcallingJAVA/ASM/gvbjwait.s [hunk 8]
```diff
@@ -113,6 +112,7 @@ PAADDR1  DS    D
 PAADDR2  DS    D
```
Comment (KipTwitchell) [inline POC-iterations/ASMcallingJAVA/ASM/gvbjwait.s:112]: comments

**Review by KipTwitchell — CHANGES_REQUESTED** [review]
See comments.

**Comment by neilbeesley** [comment]
Changes made.

File: POC-iterations/ASMcallingJAVA/ASM/gvbjmain.s [hunk 9]
```diff
@@ -125,114 +126,131 @@ GVBJMAIN CSECT
...
+MAIN_095 EQU   *
+         sam64
+         sysstate amode64=YES
 *
 ***********************************************************************
 * INITIALIZE CTT AREA                                                 *
 ***********************************************************************
-*
-*
 MAIN_096 EQU   *
```
[the remark this answers is not in the document]
Comment (KipTwitchell) [reply POC-iterations/ASMcallingJAVA/ASM/gvbjmain.s:182]: no change needed.

**Review by KipTwitchell — APPROVED** [review]

**Pull request merged**